zyte_common_items/items.py

```diff
@@ -81,9 +89,10 @@ class ArticleFromList(Item):
     #: All images.
     images: Optional[List[Image]] = None
 
-    #: The probability (0 for 0%, 1 for 100%) that this record represents an
-    #: actual article.
-    probability: Optional[float] = None
+    #: Data extraction process metadata.
+    metadata: Optional[ProbabilityMetadata] = attrs.field(
+        default=None, converter=attrs.converters.optional(MetadataCaster(ProbabilityMetadata)), kw_only=True  # type: ignore
```

Comment:
Does the conversion work when list item metadata is accessed as a part of the `@field` in the page object, not as a part of the final item?

Reply:
`test_items.py::test_metadata_generic` accesses the field directly.

Reply:
It checks that product_page.metadata field works correctly, but not product_list_page.products[0].metadata.

Reply:
Yeah, looks good! I actually was a bit wrong asking this; my worry was that there is a convertor without a corresponding processor. But we don't have an Extractor defined for those list items, so there are no processors at all.